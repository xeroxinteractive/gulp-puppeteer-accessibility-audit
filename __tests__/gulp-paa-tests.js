'use strict';

jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;

const gutil = require('gulp-util');

let paa;
let stream;
let fixture;

beforeEach(async () => {
  paa = require('../');

  fixture = new gutil.File({
    base: __dirname + '/',
    cwd: __dirname,
    path: __dirname + '/fixture.html'
  });

  stream = paa({
    puppeteerConfig: {
      headless: false
    }
  });
});

test('should return a report', (done) => {

  stream.on('data', function (file) {
    expect(file).toHaveProperty("paa");
    expect(file.paa).toHaveProperty("audit");
    expect(file.paa).toHaveProperty("report");
  });

  stream.on('end', function () {
    done();
  });

  stream.write(fixture);
  stream.end();

});

test('should format a report', (done) => {

  let report = paa.reporter(function (output) {
    expect(output).not.toBeNull();
    expect(output).toMatchSnapshot();
  });

  stream.on('end', function () {
    done();
  });

  stream.write(fixture);
  stream.pipe(report);
  stream.end();
});

test('Should audit the file and throw  at the first error', (done) => {

  let fail = paa.failOnError();
  let thrown = false;

  stream.on('data', function (file) {
    expect(file).toHaveProperty("paa");
    expect(file.paa).toHaveProperty("audit");
    expect(file.paa).toHaveProperty("report");

    expect(file.paa.audit).toMatchSnapshot();
    expect(file.paa.report).toMatchSnapshot();
  });

  fail.on('error', function (error) {
    thrown = true;
    expect(error).not.toBeNull();
    expect(error).toMatchSnapshot();
    done();
  });

  fail.on('end', function () {
    expect(thrown).toBe(true);
    done();
  });

  stream.write(fixture);
  stream.pipe(fail);
  stream.end();
});

test('Should audit the file and throw at the end', (done) => {
  
    let fail = paa.failAfterError();
    let thrown = false;
  
    stream.on('data', function (file) {
      expect(file).toHaveProperty("paa");
      expect(file.paa).toHaveProperty("audit");
      expect(file.paa).toHaveProperty("report");
  
      expect(file.paa.audit).toMatchSnapshot();
      expect(file.paa.report).toMatchSnapshot();
    });
  
    fail.on('error', function (error) {
      thrown = true;
      expect(error).not.toBeNull();
      expect(error).toMatchSnapshot();
      done();
    });
  
    fail.on('end', function () {
      expect(thrown).toBe(true);
      done();
    });
  
    stream.write(fixture);
    stream.pipe(fail);
    stream.end();
  });