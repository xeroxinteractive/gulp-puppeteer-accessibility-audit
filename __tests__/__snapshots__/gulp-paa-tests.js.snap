// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should audit the file and throw  at the first error 1`] = `
Array [
  Object {
    "code": "AX_ARIA_12",
    "elements": "",
    "heading": "This element does not support ARIA roles, states and properties",
    "result": "PASS",
    "severity": "Warning",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_12",
  },
  Object {
    "code": "AX_ARIA_06",
    "elements": "",
    "heading": "aria-owns should not be used if ownership is implicit in the DOM",
    "result": "NA",
    "severity": "Warning",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_06",
  },
  Object {
    "code": "AX_ARIA_09",
    "elements": "",
    "heading": "Elements with ARIA roles must be in the correct scope",
    "result": "PASS",
    "severity": "Severe",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_09",
  },
  Object {
    "code": "AX_AUDIO_01",
    "elements": "",
    "heading": "Audio elements should have controls",
    "result": "PASS",
    "severity": "Warning",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_audio_01",
  },
  Object {
    "code": "AX_ARIA_11",
    "elements": "",
    "heading": "This element has an invalid ARIA attribute",
    "result": "PASS",
    "severity": "Warning",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_11",
  },
  Object {
    "code": "AX_ARIA_04",
    "elements": "
DIV:nth-of-type(3) > INPUT
DIV:nth-of-type(5) > INPUT
#aria-invalid",
    "heading": "ARIA state and property values must be valid",
    "result": "FAIL",
    "severity": "Severe",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_04",
  },
  Object {
    "code": "AX_ARIA_01",
    "elements": "
DIV:nth-of-type(11) > SPAN",
    "heading": "Elements with ARIA roles must use a valid, non-abstract ARIA role",
    "result": "FAIL",
    "severity": "Severe",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_01",
  },
  Object {
    "code": "AX_TEXT_01",
    "elements": "
DIV > INPUT
DIV:nth-of-type(12) > DIV:nth-of-type(3) > INPUT
LABEL > INPUT",
    "heading": "Controls and media elements should have labels",
    "result": "FAIL",
    "severity": "Severe",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_text_01",
  },
  Object {
    "code": "AX_HTML_02",
    "elements": "
#forInput
#forInput",
    "heading": "Any ID referred to via an IDREF must be unique in the DOM",
    "result": "FAIL",
    "severity": "Severe",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_html_02",
  },
]
`;

exports[`Should audit the file and throw  at the first error 2`] = `
"*** Begin accessibility audit results ***
An accessibility audit found 
Errors:
Error: AX_ARIA_04 (ARIA state and property values must be valid) failed on the following elements (1 - 3 of 3):
DIV:nth-of-type(3) > INPUT
DIV:nth-of-type(5) > INPUT
#aria-invalid
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_04 for more information.

Error: AX_ARIA_01 (Elements with ARIA roles must use a valid, non-abstract ARIA role) failed on the following element:
DIV:nth-of-type(11) > SPAN
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_01 for more information.

Error: AX_TEXT_01 (Controls and media elements should have labels) failed on the following elements (1 - 3 of 3):
DIV > INPUT
DIV:nth-of-type(12) > DIV:nth-of-type(3) > INPUT
LABEL > INPUT
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_text_01 for more information.

Error: AX_HTML_02 (Any ID referred to via an IDREF must be unique in the DOM) failed on the following elements (1 - 2 of 2):
#forInput
#forInput
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_html_02 for more information.


*** End accessibility audit results ***"
`;

exports[`Should audit the file and throw at the end 1`] = `
Array [
  Object {
    "code": "AX_ARIA_12",
    "elements": "",
    "heading": "This element does not support ARIA roles, states and properties",
    "result": "PASS",
    "severity": "Warning",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_12",
  },
  Object {
    "code": "AX_ARIA_06",
    "elements": "",
    "heading": "aria-owns should not be used if ownership is implicit in the DOM",
    "result": "NA",
    "severity": "Warning",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_06",
  },
  Object {
    "code": "AX_ARIA_09",
    "elements": "",
    "heading": "Elements with ARIA roles must be in the correct scope",
    "result": "PASS",
    "severity": "Severe",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_09",
  },
  Object {
    "code": "AX_AUDIO_01",
    "elements": "",
    "heading": "Audio elements should have controls",
    "result": "PASS",
    "severity": "Warning",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_audio_01",
  },
  Object {
    "code": "AX_ARIA_11",
    "elements": "",
    "heading": "This element has an invalid ARIA attribute",
    "result": "PASS",
    "severity": "Warning",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_11",
  },
  Object {
    "code": "AX_ARIA_04",
    "elements": "
DIV:nth-of-type(3) > INPUT
DIV:nth-of-type(5) > INPUT
#aria-invalid",
    "heading": "ARIA state and property values must be valid",
    "result": "FAIL",
    "severity": "Severe",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_04",
  },
  Object {
    "code": "AX_ARIA_01",
    "elements": "
DIV:nth-of-type(11) > SPAN",
    "heading": "Elements with ARIA roles must use a valid, non-abstract ARIA role",
    "result": "FAIL",
    "severity": "Severe",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_01",
  },
  Object {
    "code": "AX_TEXT_01",
    "elements": "
DIV > INPUT
DIV:nth-of-type(12) > DIV:nth-of-type(3) > INPUT
LABEL > INPUT",
    "heading": "Controls and media elements should have labels",
    "result": "FAIL",
    "severity": "Severe",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_text_01",
  },
  Object {
    "code": "AX_HTML_02",
    "elements": "
#forInput
#forInput",
    "heading": "Any ID referred to via an IDREF must be unique in the DOM",
    "result": "FAIL",
    "severity": "Severe",
    "url": "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_html_02",
  },
]
`;

exports[`Should audit the file and throw at the end 2`] = `
"*** Begin accessibility audit results ***
An accessibility audit found 
Errors:
Error: AX_ARIA_04 (ARIA state and property values must be valid) failed on the following elements (1 - 3 of 3):
DIV:nth-of-type(3) > INPUT
DIV:nth-of-type(5) > INPUT
#aria-invalid
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_04 for more information.

Error: AX_ARIA_01 (Elements with ARIA roles must use a valid, non-abstract ARIA role) failed on the following element:
DIV:nth-of-type(11) > SPAN
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_01 for more information.

Error: AX_TEXT_01 (Controls and media elements should have labels) failed on the following elements (1 - 3 of 3):
DIV > INPUT
DIV:nth-of-type(12) > DIV:nth-of-type(3) > INPUT
LABEL > INPUT
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_text_01 for more information.

Error: AX_HTML_02 (Any ID referred to via an IDREF must be unique in the DOM) failed on the following elements (1 - 2 of 2):
#forInput
#forInput
See https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_html_02 for more information.


*** End accessibility audit results ***"
`;

exports[`Should audit the file and throw at the end 3`] = `
[gulp-puppeteer-accessibility-audit: 

  [33mC:\\Users\\Z269CJBB\\Documents\\GitHub\\gulp-puppeteer-accessibility-audit\\__tests__/fixture.html[39m

  [31m√ó[39m ARIA state and property values must be valid

  DIV:nth-of-type(3) > INPUT
  DIV:nth-of-type(5) > INPUT
  #aria-invalid

  [31m√ó[39m Elements with ARIA roles must use a valid, non-abstract ARIA role

  DIV:nth-of-type(11) > SPAN

  [31m√ó[39m Controls and media elements should have labels

  DIV > INPUT
  DIV:nth-of-type(12) > DIV:nth-of-type(3) > INPUT
  LABEL > INPUT

  [31m√ó[39m Any ID referred to via an IDREF must be unique in the DOM

  #forInput
  #forInput

]
`;
